src/aws/flb_aws_util.c

```diff
@@ -67,7 +67,8 @@ char *flb_aws_endpoint(char* service, char* region)
     snprintf(endpoint, len, AWS_SERVICE_ENDPOINT_FORMAT, service, region);
 
     if (is_cn) {
-        strncat(endpoint, ".cn", 3);
+        memcpy(endpoint + (len - 4), ".cn", 3);
```

Comment:
on this case, if you want to write "after" the last valid byte of ```endpoint```, I suggest you check the return value of snprintf() which returns the number of bytes written to endpoint, then you can do something like:

```c
bytes = snprintf(endpoint...)

memcpy(endpoint + bytes, ".cn", 3)
len = bytes + 3;
endpoint[len] = '\0'
```